sdks/go/pkg/beam/core/runtime/exec/plan.go

```diff
@@ -32,6 +32,8 @@ type Plan struct {
 	units []Unit
 
 	status Status
+
+	// TODO: there can be more than 1 DataSource in a bundle.
```

Comment:
<!--new_thread; commit:266f7ad642d50d8632699e01168fe7e359f60835; resolved:0-->
I assume multiple DataSources means there are side inputs?

Reply:
Not necessarily. Two DataSources can also just be flattened together, for example.

Reply:
For posterity: it seems I'm wrong here. Pipeline models use exactly one DataSource. (For non-model use of plan, we may still have multiple Impulse roots.)